packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -76,21 +101,45 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
 
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    if(_testingWithSkiaGold()) {
+      // Library prefix notation for Skia Gold.
+      golden = _addPrefix(golden);
+      // Always updating for SkiaGold testing.
+      await update(golden, imageBytes);
+    }
+
     final File goldenFile = _getGoldenFile(golden);
     if (!goldenFile.existsSync()) {
       throw TestFailure('Could not be compared against non-existent file: "$golden"');
     }
-    final List<int> goldenBytes = await goldenFile.readAsBytes();
-    // TODO(tvolkert): Improve the intelligence of this comparison.
-    if (goldenBytes.length != imageBytes.length) {
-      return false;
-    }
-    for (int i = 0; i < goldenBytes.length; i++) {
-      if (goldenBytes[i] != imageBytes[i]) {
+
+    if (_testingWithSkiaGold()) {
+      // Testing with SkiaGoldClient.
+      final bool authorized = await _skiaClient.auth(fs.directory(basedir));
+      if(!authorized) {
+        throw TestFailure('Could not authorize goldctl.');
+      }
+
+      await _skiaClient.imgtestInit();
+      return await _skiaClient.imgtestAdd(golden.path, goldenFile);
+
+    } else if (platform.isLinux) {
```

Comment:
Since the branch above returns, you don't need the else. You can just do another if below.